pkg/cache/snapshot_test.go

```diff
@@ -1331,3 +1204,166 @@ func TestSnapshotAddRemoveWorkloadWithLendingLimit(t *testing.T) {
 		})
 	}
 }
+
+func TestResourceGroups(t *testing.T) {
+	cache := New(utiltesting.NewFakeClient())
+
+	clusterQueue := utiltesting.MakeClusterQueue("test-clusterqueue").
+		ResourceGroup(
+			utiltesting.MakeFlavorQuotas("on-demand").
+				ResourceQuotaWrapper(corev1.ResourceCPU).NominalQuota("0").Append().
+				ResourceQuotaWrapper(corev1.ResourceMemory).NominalQuota("0").Append().
+				FlavorQuotas,
```

Comment:
Can we use Obj() here?

Reply:
Done. In general, I wonder if it may be more ergonomic to replace all usages of Obj() with FlavorQuotas (and also replace .Obj() all wrapper types), as I think most of the time we actually need the underlying type, and not a pointer.

Out of the scope of this PR obviously.